db/compaction/compaction_picker.cc

```diff
@@ -52,7 +59,9 @@ bool FindIntraL0Compaction(const std::vector<FileMetaData*>& level_files,
     compact_bytes += static_cast<size_t>(level_files[span_len]->fd.file_size);
     compensated_compact_bytes += level_files[span_len]->compensated_file_size;
     new_compact_bytes_per_del_file = compact_bytes / span_len;
-    if (level_files[span_len]->being_compacted ||
+    if ((level_files[0]->fd.smallest_seqno == level_files[0]->fd.largest_seqno &&
+      level_files[span_len]->fd.largest_seqno >= oldest_mem_seqno) ||
```

Comment:
Is this check necessary? My understanding is that `level_files` is sorted in descending `largest_seqno` order. So if the check at the start of this method succeeded, the check here must also be true. Am I missing something?

I wasn't aware that intra-L0 compactions always started with the newest sstable in L0. Rather than the approach take here, could we instead skip the prefix of `level_files` that contains sstables with `largest_seqno >= oldest_mem_seqno` and then build a span from there? Cc @ajkr.

Reply:
Yes， it is not necessary. I am just worried about other issues make it not keep order.

Reply:
Sorry,  I make a mistake. My intention is to check every file, rather than `level_files[0]`